### PR TITLE
[datadog_synthetics_test] Add support for grpc subtype

### PR DESCRIPTION
**Pull request by AlaricCalmette**

### CODE REVIEW

File: docs/resources/synthetics_test.md [hunk 1]
```diff
@@ -446,6 +446,7 @@ Optional:
 - `number_of_packets` (Number) Number of pings to use per test for ICMP tests (`subtype = "icmp"`) between 0 and 10.
 - `port` (Number) Port to use when performing the test.
 - `servername` (String) For SSL tests, it specifies on which server you want to initiate the TLS handshake, allowing the server to present one of multiple possible certificates on the same IP address and TCP port number.
+- `service` (String) For gRPC tests, service to target for healthcheck.
```
Comment (cswatt) [inline docs/resources/synthetics_test.md:449]:
```suggestion
- `service` (String) For gRPC tests, service to target for health check.
```

File: docs/resources/synthetics_test.md [hunk 2]
```diff
@@ -732,6 +733,7 @@ Optional:
...
```
Comment (cswatt) [inline docs/resources/synthetics_test.md:736]:
```suggestion
- `service` (String) For gRPC tests, service to target for health check.
```

**Review by cswatt — APPROVED** [review]
Leaving very nitpicky edit; approving so as not to block.

File: docs/resources/synthetics_test.md (hunk 1 above)
Comment (AlaricCalmette) [reply docs/resources/synthetics_test.md:449]: I purposefully put healthcheck to match what we have in the UI. If we're to change that's a discussion I need to have with my PMs.

**Review by therve — APPROVED** [review]

**Pull request merged**